### PR TITLE
Viatra 2.0

### PR DESCRIPTION
**Pull request by imbur**
Updating master branch to use Viatra 2.0

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,14 +1,12 @@
 // Tell Jenkins how to build projects from this repository
 pipeline {
-	agent {
-		label 'java7'
-	} 
+	agent any 
 	parameters {
-		booleanParam(defaultValue: true, description: '''This parameter is used to allow not to execute Sonar analysis. It is safe to always make this true, as the Sonar-trigger job will trigger this job without the SKIP_SONAR parameter set daily.''', name: 'SKIP_SONAR')
```
Comment (abelhegedus) [inline Jenkinsfile]: Do not remove this parameter.

File: plugins/hu.bme.mit.massif.simulink.incquery/plugin.xml [hunk 2]
```diff
@@ -1,25 +1,4 @@
 <?xml version="1.0" encoding="UTF-8"?><plugin>
-  <extension id="extension.derived.hu.bme.mit.massif.models.simulink.derived.enabler" point="org.eclipse.viatra.query.runtime.base.wellbehaving.derived.features">
```
Comment (abelhegedus) [inline plugins/hu.bme.mit.massif.simulink.incquery/plugin.xml]: This should not be deleted.

File: plugins/hu.bme.mit.massif.simulink.incquery.validation/plugin.xml [hunk 3]
```diff
@@ -1,8 +1 @@
-<?xml version="1.0" encoding="UTF-8"?><plugin>
```
Comment (abelhegedus) [inline plugins/hu.bme.mit.massif.simulink.incquery.validation/plugin.xml]: This should not be deleted.

File: setup/hu.bme.mit.massif.setup/Massif.setup [hunk 4]
```diff
@@ -578,7 +578,7 @@
             versionRange="[2.9.0,3.0.0)"/>
```
Comment (abelhegedus) [inline setup/hu.bme.mit.massif.setup/Massif.setup]: Did you check whether any other changes are necessary in the Oomph setup (on the master stream)?

Comment (imbur) [reply setup/hu.bme.mit.massif.setup/Massif.setup]: No, not really (yet) - is it sufficient to go through the install process using the installer to see if any further changes are needed?

Comment (abelhegedus) [reply setup/hu.bme.mit.massif.setup/Massif.setup]: Yes.

File: Jenkinsfile [hunk 5]
```diff
@@ -1,14 +1,13 @@
...
-        choice(choices: 'ci\nintegration\nrelease', description: '', name: 'BUILD_TYPE')
+		string(name: 'VIATRA_REPOSITORY', defaultValue: 'https://hudson.eclipse.org/viatra/job/viatra-master/lastSuccessfulBuild/artifact/releng/org.eclipse.viatra.update/target/repository')
```
Comment (ujhelyiz) [inline Jenkinsfile]: Is this parameter required for master builds? I'd hazard no - we should rely on the latest VIATRA version instead...

File: Jenkinsfile [hunk 6]
```diff
@@ -18,19 +17,17 @@ pipeline {
...
             steps {
-                wrap([$class: 'TimestamperBuildWrapper']) {
-                    configFileProvider([
-                        configFile(fileId: 'maven-toolchain-oldjdk', variable: 'TOOLCHAIN'),
-                        configFile(fileId: 'default-maven-settings', variable: 'MAVEN_SETTINGS')]) {
-                            sh "mvn clean verify -B -t $TOOLCHAIN -s $MAVEN_SETTINGS -f releng/hu.bme.mit.massif.parent/pom.xml -Dmaven.repo.local=$WORKSPACE/.repository -DBUILD_TYPE=${params.BUILD_TYPE}"
-                            sh "./releng/massif.commandevaluation.server-package/prepareMatlabServerPackage.sh"
-                        }
+                configFileProvider([configFile(fileId: 'default-maven-toolchains', variable: 'TOOLCHAIN'), configFile(fileId: 'default-maven-settings', variable: 'MAVEN_SETTINGS')]) {
+                    sh "mvn clean install -B -t $TOOLCHAIN -s $MAVEN_SETTINGS -f releng/hu.bme.mit.massif.parent/pom.xml -Dviatra.repository.url=${params.VIATRA_REPOSITORY} -Dmaven.repo.local=$WORKSPACE/.repository"
```
Comment (ujhelyiz) [inline Jenkinsfile]: Again, do not add the VIATRA repository here.

File: releng/hu.bme.mit.massif.import.parent/pom.xml [hunk 7]
```diff
@@ -14,8 +14,8 @@
 
 	</modules>
 	<properties>
-		<tycho.version>0.18.0</tycho.version>
-		<xtend.compiler.version>2.4.2</xtend.compiler.version>
+		<tycho.version>2.0.0</tycho.version>
```
Comment (ujhelyiz) [inline releng/hu.bme.mit.massif.import.parent/pom.xml]: This is almost certainly incorrect. The latest tycho release is 1.2.0 (AFAIK).

File: Jenkinsfile (hunk 5 above)
Comment (imbur) [reply Jenkinsfile]: I guess this means the latest VIATRA version (last successful build). Did you mean latest released version of VIATRA?

File: releng/hu.bme.mit.massif.parent/pom.xml [hunk 8]
```diff
@@ -50,24 +56,31 @@
...
 
 		
 	</modules>
 	<properties>
 		<tycho.version>1.0.0</tycho.version>
 		<xtend.compiler.version>2.12.0</xtend.compiler.version>
-		<viatra.compiler.version>1.6.0</viatra.compiler.version>
-		<viatra.repository.url>http://download.eclipse.org/viatra/updates/release/</viatra.repository.url>
-		<viatra.maven.repository.url>https://repo.eclipse.org/content/repositories/viatra-releases/</viatra.maven.repository.url>
+		<viatra.compiler.version>2.0.0-SNAPSHOT</viatra.compiler.version>
```
Comment (ujhelyiz) [inline releng/hu.bme.mit.massif.parent/pom.xml]: Do not use snapshot version of VIATRA in the master branch.

File: releng/hu.bme.mit.massif.parent/pom.xml [hunk 9]
```diff
@@ -50,24 +56,31 @@
...
 		<module>../../maven/massif-communication-runtime</module>
 		<module>../../maven/massif-matlabcontrol-runtime</module>
 		<module>../../maven/massif-commandevaluation-runtime</module>
+		<module>../../maven/massif-simulink-emf</module>
 
 		
 	</modules>
 	<properties>
 		<tycho.version>1.0.0</tycho.version>
 		<xtend.compiler.version>2.12.0</xtend.compiler.version>
```
Comment (ujhelyiz) [inline releng/hu.bme.mit.massif.parent/pom.xml]: Make sure a consistent Xtend version is used througout the project

File: releng/hu.bme.mit.massif.import.parent/pom.xml [hunk 10]
```diff
@@ -14,8 +14,8 @@
...
+		<xtend.compiler.version>2.13.0</xtend.compiler.version>
```
Comment (ujhelyiz) [inline releng/hu.bme.mit.massif.import.parent/pom.xml]: Select Xtend version consistently.

File: Jenkinsfile (hunk 5 above)
Comment (abelhegedus) [reply Jenkinsfile]: This parameter was introduced to allow testing specific VIATRA builds (even those created Gerrit changes) without modifying Massif code (specifically cyclic linking problems at some point).

File: releng/hu.bme.mit.massif.import.parent/pom.xml (hunk 7 above)
Comment (imbur) [reply releng/hu.bme.mit.massif.import.parent/pom.xml]: Based on [this list](https://wiki.eclipse.org/Tycho/Release_Notes), the latest release is 1.1.0 - so I'll put that in.

File: releng/hu.bme.mit.massif.parent/pom.xml (hunk 9 above)
Comment (imbur) [reply releng/hu.bme.mit.massif.parent/pom.xml]: OK, I'll update this to 2.13.0

File: releng/hu.bme.mit.massif.import.parent/pom.xml (hunk 7 above)
Comment (ujhelyiz) [reply releng/hu.bme.mit.massif.import.parent/pom.xml]: Tycho 1.2 is out (http://search.maven.org/#artifactdetails%7Corg.eclipse.tycho%7Ctycho%7C1.2.0%7Cpom and https://wiki.eclipse.org/Tycho/Release_Notes/1.2) - feel free to use it.

Comment (imbur) [reply releng/hu.bme.mit.massif.import.parent/pom.xml]: OK, thank you!

**Pull request merged**